### ISSUE TITLE
[grep.py] weechat.look.color_nicks_number is obsolete

### ISSUE THREAD

**Issue by azrdev** [weechat/scripts#230]
grep.py gives me:
~~~
14:39 python: stdout/stderr: Traceback (most recent call last):
14:39 python: stdout/stderr:   File "/home/user/.weechat/python/autoload/grep.py", line 1033, in grep_file_callback
14:39 python: stdout/stderr:     buffer_update()
14:39 python: stdout/stderr:   File "/home/user/.weechat/python/autoload/grep.py", line 1139, in buffer_update
14:39 python: stdout/stderr:     prnt_date_tags(buffer, 0, 'no_highlight', format_line(line))
14:39 python: stdout/stderr:   File "/home/user/.weechat/python/autoload/grep.py", line 1100, in format_line
14:39 python: stdout/stderr:     nick_c = color_nick(nick)
14:39 python: stdout/stderr:   File "/home/user/.weechat/python/autoload/grep.py", line 386, in color_nick
14:39 python: stdout/stderr:     idx = (sum(map(ord, nick))%color_nicks_number) + 1
14:39 python: stdout/stderr: ZeroDivisionError: long division or modulo by zero
14:39 =!= python: Fehler in Funktion "grep_file_callback"
~~~

quick glance at the code shows it tries to read that zero from `weechat.look.color_nicks_number`, which doesn't exist (anymore). This seems to be the same as #51/#92.

This is weechat 1.8, grep.py 0.7.7

**Comment by flashcode** [comment]
I didn't try the script, but according to the code, it should use this option only if the info "irc_nick_color" is missing. This info is currently deprecated, but still exists (for compatibility). Therefore, it should never use the option "weechat.look.color_nicks_number".
You should only have this problem if the irc plugin is not loaded.

**Comment by azrdev** [comment]
looked at weechat 1.8 and 1.9 both with the irc plugin, they don't have `irc_nick_color`.

**Comment by flashcode** [comment]
The info "irc_nick_color" is still present in these versions, and still in development version.
If ever I remove it in future, I'll ping the users before doing it.

**Comment by flashcode** [comment]
To check, you can try: `/eval -n ${info:irc_nick_color,nick}test`

**Comment by azrdev** [comment]
huh, okay, I looked in /set :)

Trying a little printf-debugging on the relevant code …
~~~python
    nick_color = weechat.info_get('irc_nick_color', nick)
    if not nick_color:
        # probably we're in WeeChat 0.3.0
        #debug('no irc_nick_color')
        color_nicks_number = config_int('weechat.look.color_nicks_number')
        idx = (sum(map(ord, nick))%color_nicks_number) + 1
~~~
… I found out that it doesn't get a `nick_color`, because `nick` is the empty string somehow.
Appending `or 1` to the `config_int()` line apparently fixes the whole bug.

**Comment by flashcode** [comment]
It should be fixed (version 0.7.8).

**Comment by azrdev** [comment]
It is, thank you very much!
